### PR TITLE
ci: check pyproject version before push

### PR DESCRIPTION
**Pull request by jjenscodee**
## Description

- #57 

This PR solves the issue by using `check-version` action to check if the `push-to-pypi` is needed.

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

**Review by eeberhard — APPROVED** [review]

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -8,8 +8,22 @@ on:
...
+    outputs:
+      has_changed: ${{ steps.check.outputs.has_changed }}
+      version: ${{ steps.versions.outputs.new_version }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: aica-technology/.github/.github/actions/check-version@v0.11.0
+        id: check
+        with:
+          package_path: ./python/pyproject.toml
```
Comment (domire8) [inline .github/workflows/publish-to-pypi.yml:22]: This doesn't work unfortunately, the action was designed for ROS packages that have the version inside a `<version>` tag in an xml file. You'll have to parse out the version yourself, like I did in modulo, I'm afraid

Comment (eeberhard) [reply .github/workflows/publish-to-pypi.yml:22]: oops I approved too fast

**Review by domire8 — APPROVED** [review]
Looks good to me now, I assume you tested it?

**Comment by jjenscodee** [comment]
yeah I test it here:
https://github.com/aica-technology/api/actions/runs/6613220204/job/17960639141
and here
https://github.com/aica-technology/api/actions/runs/6613242149/job/17960703179

**Review by eeberhard — APPROVED** [review]
Now that it's tested I will approve it again 😁

**Pull request merged**